src/frontend/src/session/cursor_manager.rs

```diff
@@ -313,20 +344,30 @@ impl SubscriptionCursor {
             )
             .into());
         }
-        // `FETCH NEXT` is equivalent to `FETCH 1`.
-        if count != 1 {
-            Err(crate::error::ErrorCode::InternalError(
-                "FETCH count with subscription is not supported".to_string(),
-            )
-            .into())
-        } else {
-            let (row, pg_descs) = self.next_row(handle_args).await?;
-            if let Some(row) = row {
-                Ok((vec![row], pg_descs))
-            } else {
-                Ok((vec![], pg_descs))
+
+        let mut ans = Vec::with_capacity(std::cmp::min(100, count) as usize);
+        let mut cur = 0;
+        let mut pg_descs_ans = vec![];
+        while cur < count {
+            let pg_descs = self.seek_descs(&handle_args).await?;
+            if pg_descs_ans.is_empty() {
+                pg_descs_ans = pg_descs;
+            } else if !pg_descs_ans.eq(&pg_descs) {
+                break;
+            }
+            let (row, _) = self.next_row(&handle_args).await?;
```

Comment:
The newly added `seek_desc` as well as `cache_seek_row` and `cache_seek_descs` in the `SubscriptionCursor` struct seems too ad-hoc to me. How about keeping the original implementation but pass a `expected_pg_decs` as a parameter to `next_row` instead?